NEWS.md

```diff
@@ -40,6 +40,8 @@ shiny 1.5.0.9000
 
 * Fixed #1942: Calling `runApp("app.R")` no longer ignores options passed into `shinyApp()`. This makes it possible for Shiny apps to specify what port/host should be used by default. (#2969)
 
+* Fixed #3033: When a `DiskCache` was created with both `max_n` and `max_size`, too many items could get pruned when `prune()` was called. (#3034)
```

Comment:
```suggestion
* Fixed #3033: When a `DiskCache` was created with both `max_n` and `max_size`, `prune()` didn't always prune objects that it should've. (#3034)
```

Reply:
Oh, actually, now that I think about it again, the original news item was correct. If only `max_n` or `max_size` was used, it would remove the correct items. It's only after the removal that the `info` data frame was modified incorrectly, and that's a problem only if `info` is used again. That would only happen if `max_n` and `max_size` were both used.

(Note that the first step of pruning, for `max_age`, did not record the sucessfully removed files -- it just assumed that the removal worked -- so it wasn't a problem there. But I have added the check for successful removal in that stage in this PR.)

I'm going to change the news back to reflect this.